src/types.ts

```diff
@@ -178,10 +215,10 @@ export interface CellStyle {
   selectedCellClass?: string;
 }
 
-export interface Components {
+export interface Component {
   key: string;
-  content?: StylingLayout;
-  header?: StylingLayout;
+  content?: ContentStyling;
+  header?: HeaderStyling;
```

Comment:
These are not the types the `styleObj` has in `getBaseStyling`. I intentionally made the the same type so that you wouldn't need a union to type `getBaseStyling`

Reply:
but `StylingLayout` is not the same as `ContentStyling` and `HeaderStyling`. Does that work even though they do not  have anything to do with eachother?

Reply:
Good question. I see your point.
<img width="613" alt="Screenshot 2022-08-22 at 11 26 57" src="https://user-images.githubusercontent.com/4471489/185888105-545d849d-74da-4f69-ad9f-ae469ca5900f.png">
 It works. As of now all properties in ContentStyling, HeaderStyling, and StylingLayout are optional. We probably want to change it.

Reply:
We need to think about whether to use optional or undefined in types.ts. 
For this specific case, I think they can be optional, as it is okay not to provide any value.

Reply:
Yes they are actually optional. Hmm I think either we do one generic as before just `StylingLayout` or we use yours and then set the type for getBaseStyle to `styleObj: ContentStyling | HeaderStyling`. I made it just one type since they overlap a lot but everything is optional so you could use one type for both

Reply:
I would like to match Component in src/qae/object-properties.js with Component in src/types.ts. So I choose to `styleObj: ContentStyling | HeaderStyling`

Reply:
sounds good

Reply:
Dona,  please have a look